components/admin_console/team_channel_settings/users_to_be_removed_modal.tsx

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+import {Modal} from 'react-bootstrap';
+
+import {UserProfile} from '@mattermost/types/users';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+
+import GroupsUsers from './group/group_users';
+
+type Props = {
+
+    /*
+     * Number of users to be removed
+     */
+    total: number;
+
+    /*
+     * Scope that the users are to be removed from
+     * can be either 'team' or 'channel'
+     */
+    scope: 'team' | 'channel';
+
+    /*
+     * Id of the scope that the users are to be removed from
+     * can be either a Team ID or a Channel ID
+     */
+    scopeId: string;
+
+    /*
+     * users to be removed
+     */
+    users: UserProfile[];
+
+    onExited?: () => void;
+};
+
+const UsersToBeRemovedModal = ({total, scope, scopeId, users, onExited}: Props) => {
+    const [show, setShow] = useState(true);
+
+    const handleHide = () => setShow(false);
```

Comment:
The main time you need it is when you're passing a method to another React component. Without `useCallback`, you're declaring a new function each time this component is rendered, and because you're passing a new function into the child component, it is also guaranteed to re-render. So in this case, every time UsersToBeRemovedModal re-renders, the Modal is guaranteed to re-render even if nothing visible changed.

The main time it's not needed is if we're passing the function directly into a plain HTML component since there's nothing to re-render there. You could still use `useCallback` there, but I don't think it makes any difference performance-wise

Reply:
Thanks @hmhealey I also found this [article](https://dmitripavlutin.com/dont-overuse-react-usecallback/) about how to effectively use useCallback